server/src/main/java/io/crate/metadata/doc/DocIndexMetadata.java

```diff
@@ -852,4 +860,25 @@ public Version versionUpgraded() {
     public boolean isClosed() {
         return closed;
     }
+
+    /**
+     * Even though referenced marked as "dropped" are excluded from {@link #columns}, when a sub-column
+     * of an object col is dropped, the {@link ObjectType#innerTypes()} still contain those dropped sub-columns.
+     */
+    @SuppressWarnings("unchecked")
+    private DataType<?> removeDroppedColsFromInnerTypes(Map<String, Object> columnProperties, DataType<?> type) {
+        if (type.id() == ObjectType.ID) {
+            ObjectType.Builder builder = new ObjectType.Builder();
```

Comment:
don't we need special handling for arrays? dropped object can have array in some deep nested level

Reply:
as mentioned, this is to be followed up